.github/workflows/build_and_run_selected_test.yml

```
@@ -0,0 +1,117 @@
+name: Build and Run Selected Test
+
+on:
+  workflow_dispatch:
+    inputs:
+      test-selection:
+        description: 'Specification of selected test(s), as accepted by pytest -k'
+        required: true
+        type: string
+      run-count:
+        description: 'Number of test runs to perform'
+        required: true
+        type: number
+      archs:
+        description: 'Archs to run tests on, e. g.: ["x64", "arm64"]'
+        default: '["x64"]'
+        required: true
+        type: string
+      build-types:
+        description: 'Build types to run tests on, e. g.: ["debug", "release"]'
+        default: '["release"]'
+        required: true
+        type: string
+      pg-versions:
+        description: 'Postgres versions to use for testing,  e.g,: [{"pg_version":"v16"}, {"pg_version":"v17"}])'
+        default: '[{"pg_version":"v17"}]'
+        required: true
+        type: string
+  pull_request:
+
+defaults:
+  run:
+    shell: bash -euxo pipefail {0}
+
+env:
+  RUST_BACKTRACE: 1
+  COPT: '-Werror'
+
+jobs:
+  meta:
+    uses: ./.github/workflows/_meta.yml
+    with:
+      github-event-name: ${{ github.event_name }}
+
+  build-and-test-locally:
+    needs: [ meta ]
+    strategy:
+      fail-fast: false
+      matrix:
+        arch: ${{ fromJson(inputs.archs) }}
+        build-type: ${{ fromJson(inputs.build-types) }}
+    uses: ./.github/workflows/_build-and-test-locally.yml
+    with:
+      arch: ${{ matrix.arch }}
+      build-tools-image: neondatabase/build-tools:pinned-bookworm
+      build-tag: ${{ needs.meta.outputs.build-tag }}
+      build-type: ${{ matrix.build-type }}
+      test-cfg: ${{ inputs.pg-versions }}
+      test-selection: ${{ inputs.test-selection }}
+      test-run-count: ${{ fromJson(inputs.run-count) }} 
+
+    secrets: inherit
+
+
+  create-test-report:
+    needs: [ build-and-test-locally ]
+    if: ${{ !cancelled() }}
+    permissions:
+      id-token: write # aws-actions/configure-aws-credentials
+      statuses: write
+      contents: write
+      pull-requests: write
+    outputs:
+      report-url: ${{ steps.create-allure-report.outputs.report-url }}
+
+    runs-on: [ self-hosted, small ]
+    container:
+      image: neondatabase/build-tools:pinned-bookworm
+      credentials:
+        username: ${{ secrets.NEON_DOCKERHUB_USERNAME }}
+        password: ${{ secrets.NEON_DOCKERHUB_PASSWORD }}
+      options: --init
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Create Allure report
+        if: ${{ !cancelled() }}
+        id: create-allure-report
+        uses: ./.github/actions/allure-report-generate
+        with:
+          store-test-results-into-db: true
+          aws-oicd-role-arn: ${{ vars.DEV_AWS_OIDC_ROLE_ARN }}
+        env:
+          REGRESS_TEST_RESULT_CONNSTR_NEW: ${{ secrets.REGRESS_TEST_RESULT_CONNSTR_NEW }}
```

I've created a new project (`rapid-feather-80427624`) for test results that we want to exclude from regular runs, and added a secret for that:

```suggestion
          REGRESS_TEST_RESULT_CONNSTR_NEW: ${{ secrets.REGRESS_TEST_RESULT_CONNSTR_DEV }}
```

We can drop `special_kind`

---

I've thought about that field just today because I wanted to clean the "Flaky tests" dashboard of unwanted code-breaking PRs. For now, I have a custom filter that I use when extracting useful information from the database (like reference NOT IN ('refs/pull/10515/merge', 'refs/pull/11202/merge', 'refs/pull/11216/merge', 'refs/pull/11347/merge', 'refs/pull/11494/merge', 'refs/pull/10591/merge')), but it's not an option for the dashboard, I think. But if we leave special_kind, then we can add a simple filter to the dashboard's queries and store in the field for such PRs something like "Code-breaking PR #10515".

---

> but it's not an option for the dashboard

We can add a filter for that to the dashboard (there's an example of it in `Neon / Regression Test Suite / Flaky Tests`)

Having it in a database either requires knowing it in advance (which in most cases is not true) or editing the data after the fact — and I would like to avoid that

---

>  I have a custom filter that I use when extracting useful information from the database (like reference NOT IN ('refs/pull/10515/merge', 'refs/pull/11202/merge', 'refs/pull/11216/merge', 'refs/pull/11347/merge', 'refs/pull/11494/merge', 'refs/pull/10591/merge')), but it's not an option for the dashboard

You can add exactly this logic to the dashboard to exclude such PRs

> thus adding extra (useful) information to the database when analyzing failures.

That definitely doesn't sound like a "special_kind"

---

I've got rid of special_kind and used the other database for results.